fs/ceph/mds_client.c

```diff
@@ -902,8 +902,17 @@ static void __register_request(struct ceph_mds_client *mdsc,
 	ceph_mdsc_get_request(req);
 	insert_request(&mdsc->request_tree, req);
 
-	req->r_uid = current_fsuid();
-	req->r_gid = current_fsgid();
+	uid_t orig_uid = from_kuid(current_user_ns(), current_uid());
```

Comment:
What is current_user_ns() here / current_uid()? What context is this invoked in?

Reply:
Maybe

    user_ns = mdsc->fsc->sb->user_ns
    req->r_uid = make_kuid(user_ns, orig_uid);
    req->r_gid = make_kgid(user_ns, orig_gid);

Reply:
Actually leave this code alone and instead 

https://elixir.bootlin.com/linux/v5.10.90/source/fs/ceph/mds_client.c#L2581

"Fix" the caller_uid as we encode the request